2021/test/RingImageInfo.java

```diff
@@ -0,0 +1,108 @@
+    private Telemetry.Item imageWidthTlm;
+    private Telemetry.Item pixelsPerInchTlm;
+    private Telemetry.Item distanceBtWWebcamAndGrabberTlm;
+    private Robot myRobot;
+    private Telemetry telemetry;
+
+
+    public RingImageInfo(Robot robot) {
+        myRobot = robot;
+        telemetry = myRobot.telemetry;
+        telemetry.setAutoClear(false);
+        //caption: what appears on the phone
+        ringPositionTlm = telemetry.addData("LeftOrigin", "unknown");
+        ringConfidTlm = telemetry.addData("Confidence", "N/A");
+        ringConfidTlm2 = telemetry.addData("Confidence2", "N/A");
+        ringTypeTlm = telemetry.addData("RingType","unknown");
+        ringTypeTlm2 = telemetry.addData("RingType2","unknown");
+        imageMidpointTlm = telemetry.addData("Image_Mdpt", "unknown");
+        ringMidpointTlm = telemetry.addData("Ring Mdpt", "unknown");
+        ringTlm = telemetry.addData("kind", "unknown");
+        deltaTlm = telemetry.addData("delta", "unknown");
+        numObjectsSeenTlm = telemetry.addData("numRings",-1);
+        pathTlm = telemetry.addData("AllianceClr", "unknown");
+        widthTlm = telemetry.addData("ringWidth", "unknown");
+        imageWidthTlm = telemetry.addData("imageWidth", -1);
+        marginTlm = telemetry.addData("margin" , "unknown");
+        pixelsPerInchTlm = telemetry.addData("pixelsPerInch", "unknown");
+        distanceBtWWebcamAndGrabberTlm = telemetry.addData("distance BtW Webcam and Grabber","unknown");
+        currLocationTlm = telemetry.addData("curr location", "in RingImageInfo constructor");
+    }
+    protected double getConfidence(){
+        return confidence;
+    }
+    protected String getRingType(){
+        return ringType;
+    }
+
+    protected void getImageInfo(RingDetectionTask.RingDetectionEvent event) {
+        //confidence is the likelihood that the object we detect is a ring in percentage
+        //get(0) = gets the first item in the list of recognition objects pointed to by rings. rings = a variable in the ring detection event
+        confidence = event.rings.get(0).getConfidence();
+        ringConfidTlm.setValue(confidence);
+        //left is the left coordinate of the ring(s)
+        left = event.rings.get(0).getLeft();
+
+        ringType = event.rings.get(0).getLabel(); //LABEL_QUAD_RINGS LABEL_SINGLE_RING
+        ringTypeTlm.setValue(ringType);
+
+        ringKind = event.kind; //OBJECTS_DETECTED
+
+        numObjectsSeen = event.rings.size();
+        numObjectsSeenTlm.setValue(numObjectsSeen);
+        currLocationTlm.setValue("in getImageInfo");
+        if (numObjectsSeen > 1){
+            confidence = event.rings.get(1).getConfidence();
+            ringConfidTlm2.setValue(confidence);
+            ringType = event.rings.get(1).getLabel(); //LABEL_QUAD_RINGS LABEL_SINGLE_RING
+            ringTypeTlm2.setValue(ringType);
+        }
+
+    }
+    //CONTINUE HERE ********
+    //ADD METHOD TO RETURN THE RING TYPE
+
+}
```

Comment:
This entire class is generic ring detection software and as such should live in terkel.  Please move it over there.